vm/devices/net/netvsp/src/lib.rs

```diff
@@ -4570,11 +4570,16 @@ impl<T: 'static + RingMem> NetChannel<T> {
                     PacketData::Init(init) => {
                         let requested_version = init.protocol_version;
                         let version = check_version(requested_version);
+                        // Normally the maximum MDL chain length is 34, which comes from the following formula:
+                        //    RNDIS Max Simulataneous Packets (8) * 3 + 10 = 34
+                        // However net_mana has a hardcoded limit of 31 transmit segments, based on a hardware limitation.
+                        // Reducing the maximum chain length forces Netvsc to bounce packets which exceed the hardware limit.
+                        let max_mdl_chain_len = 31;
```

Comment:
If you're going to limit this, get this from a method on `Endpoint` rather than hard code MANA limitations here.

Reply:
(And of course you need to enforce this somewhere.)